Src/Constraints.cpp

```diff
@@ -168,4 +168,78 @@ VolumeConstraint createVolumeConstraint(
   float targetVolume = glm::dot(glm::cross(x21, x31), x41) / 6.0f;
   return VolumeConstraint(id, k, {targetVolume}, {x1.id, x2.id, x3.id, x4.id});
 }
+
+void BendConstraintProjection::operator()(
+    const std::vector<Node>& nodes,
+    const std::array<uint32_t, 4>& nodeIds,
+    std::array<glm::vec3, 4>& projected) const {
+
+  const Node& x1 = nodes[nodeIds[0]];
+  const Node& x2 = nodes[nodeIds[1]];
+  const Node& x3 = nodes[nodeIds[2]];
+  const Node& x4 = nodes[nodeIds[3]];
+
+  glm::vec3 p2 = x2.position - x1.position;
+  glm::vec3 p3 = x3.position - x1.position;
+  glm::vec3 p4 = x4.position - x1.position;
+
+  glm::vec3 p2Xp3 = glm::cross(p2, p3);
+  glm::vec3 p2Xp4 = glm::cross(p2, p4);
+
+  glm::vec3 n1 = glm::normalize(glm::cross(p2, p3));
+  glm::vec3 n2 = glm::normalize(glm::cross(p2, p4));
+
+  float d = glm::dot(n1, n2);
+
+  float C = acos(glm::dot(n1, n2)) - this->initialAngle;
+  projected[0] = x1.position;
+  projected[1] = x2.position;
+  projected[2] = x3.position;
+  projected[3] = x4.position;
+
+  glm::vec3 q3 = (glm::cross(p2, n2) + (glm::cross(n1, p2) * d)) / glm::length(p2Xp3);
+  glm::vec3 q4 = (glm::cross(p2, n1) + (glm::cross(n2, p2) * d)) / glm::length(p2Xp4);
+  glm::vec3 q2 = -((glm::cross(p3, n2) + (glm::cross(n1, p3) * d)) / glm::length(p2Xp3))
+    -((glm::cross(p4, n1) + (glm::cross(n2, p4) * d)) / glm::length(p2Xp4));
+  glm::vec3 q1 = -q2 - q3 - q4;
+
+  float massSum = x1.mass + x2.mass + x3.mass + x4.mass;
+  float qSquaredSum = pow(glm::length(q1), 2) + pow(glm::length(q2), 2) + pow(glm::length(q3), 2) + pow(glm::length(q4), 2);
```

Comment:
When translating math equations to code, make sure to simplify the computation where possible. Here we are unnecessarily doing 4 sqrts and 4 pow(x,2) operations. Instead we just need to use dot(x,x) for each of these.